### ISSUE TITLE
Problem in helperUpdateTVMaze - deletes settings table

### ISSUE THREAD

**Issue by faulander** [faulander/P4S#14]
I am now getting the same error again

```
2020-01-22 20:07:14,012 INFO [newshows.helpers:196] Show already in DB: Sorcerous Stabber Orphen
[2020-01-22 20:07:14,742] ERROR:huey:Worker-1:Unhandled exception in task 2469f3c5-3481-441e-ae13-3835ffda4298.
Traceback (most recent call last):
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\options.py", line 581, in get_field
    return self.fields_map[field_name]
KeyError: 'page'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\huey\api.py", line 345, in _execute
    task_value = task.execute()
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\huey\api.py", line 697, in execute
    return func(*args, **kwargs)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\huey\contrib\djhuey\__init__.py", line 132, in inner
    return fn(*args, **kwargs)
  File "H:\Git\P4S\src\newshows\helpers.py", line 202, in HelperUpdateTVMaze
    q = Show.objects.filter(Q(pk=1)).update(page=page)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\query.py", line 748, in update
    query.add_update_values(kwargs)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\sql\subqueries.py", line 117, in add_update_values
    field = self.get_meta().get_field(name)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\options.py", line 583, in get_field
    raise FieldDoesNotExist("%s has no field named '%s'" % (self.object_name, field_name))
django.core.exceptions.FieldDoesNotExist: Show has no field named 'page'
2020-01-22 20:07:14,742 ERROR [huey:360] Unhandled exception in task 2469f3c5-3481-441e-ae13-3835ffda4298.
Traceback (most recent call last):
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\options.py", line 581, in get_field
    return self.fields_map[field_name]
KeyError: 'page'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\huey\api.py", line 345, in _execute
    task_value = task.execute()
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\huey\api.py", line 697, in execute
    return func(*args, **kwargs)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\huey\contrib\djhuey\__init__.py", line 132, in inner
    return fn(*args, **kwargs)
  File "H:\Git\P4S\src\newshows\helpers.py", line 202, in HelperUpdateTVMaze
    q = Show.objects.filter(Q(pk=1)).update(page=page)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\query.py", line 748, in update
    query.add_update_values(kwargs)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\sql\subqueries.py", line 117, in add_update_values
    field = self.get_meta().get_field(name)
  File "C:\Users\arogl\.virtualenvs\P4S-7tjkeV1a\lib\site-packages\django\db\models\options.py", line 583, in get_field
    raise FieldDoesNotExist("%s has no field named '%s'" % (self.object_name, field_name))
django.core.exceptions.FieldDoesNotExist: Show has no field named 'page'

```

It seems the main database is either corrupt or encrypted:

![image](https://user-images.githubusercontent.com/1115472/72885602-f6c66b00-3d53-11ea-9b5a-aaf0d1b714fe.png)

I then used the sqlite3 command line to open and check the database and all OK and working again.

The error occurs when running "run_huey", and the HelperUpdateTVMaze() runs and a page increment needs to be saved back to the database.

_Originally posted by @arogl in https://github.com/faulander/P4S/issues/13#issuecomment-577116857_

**Comment by faulander** [comment]
i have added a fix (hopefully) but i cannot test it since i am not at home.
Can you try and check if your newshows_setting table still gets deleted? If yes, it's a bug in the django orm; i double and tripple checked the code, it doesn't delete anything anywhere. I just changed that the settings gets loaded again before changing the page the 2nd time.

**Comment by arogl** [comment]
I tested be choosing my 720p profile and hitting save, no other changes. A little while later the settings had been deleted.

Default profile when going to the settings page is the "---------" profile.

**Comment by faulander** [comment]
i don't get why this gets deleted and only this table. Will continue to
look into it.

Am Do., 23. Jan. 2020 um 11:39 Uhr schrieb arogl <notifications@github.com>:

**Comment by arogl** [comment]
Could it be a database lock between the helperGetSonarrProfiles and HelperUpdateTVMaze functions?

**Comment by faulander** [comment]
I have reworked the complete settings system in 0.2.0  and it still happens, even when i don't run the management tasks in background - simply by reloading the settings page a few times. I am out of options, can't figure out why it happens. There are no errors, nothing.

I have writte the problem on stackoverflow and on the django mailing list - let's see if they come up with a solution.

**Comment by faulander** [comment]
Next thing i will try is moving to Postgres. I wanted to use sqlite to have minimum requirements, but if it works with postgres, i can live with it.

**Comment by faulander** [comment]
I think i found the problem: If i switch to Standard Django Forms, it behaves as it should, so the problem lies within Django Crispy Forms! The remodeling wasn't the baddest idea though, it's no compatible with docker and thus should receiver more users.

**Comment by faulander** [comment]
Fixed in 0.2.0
